lhotse/cut.py

```diff
@@ -1365,6 +1369,82 @@ def filter_supervisions(
         )
         return new_cut
 
+    def merge_supervisions(self) -> "MonoCut":
+        """
+        Return a copy of the cut that has all of its supervisions merged into
+        a single segment.
+
+        The new start is the start of the earliest superivion, and the new duration
+        is a minimum spanning duration for all the supervisions.
+
+        The text fields are concatenated with a whitespace, and all other string fields
+        (including IDs) are prefixed with "cat#" and concatenated with a hash symbol "#".
+        This is also applied to ``custom`` fields. Fields with a ``None`` value are omitted.
+        """
+        # "m" stands for merged in variable names below
+
+        def merge(values: Iterable[str]) -> str:
+            # e.g.
+            # values = ["1125-76840-0001", "1125-53670-0003"]
+            # return "cat#1125-76840-0001#1125-53670-0003"
+            values = list(values)
+            if len(values) == 1:
+                return values[0]
+            return "#".join(chain(["cat"], values))
+
+        sups = self.supervisions
+
+        if len(sups) == 1:
+            return self
+
+        mstart = min(s.start for s in sups)
+        mend = max(s.end for s in sups)
+        mduration = add_durations(mend, -mstart, sampling_rate=self.sampling_rate)
+
+        custom_keys = set(
+            k for s in sups if s.custom is not None for k in s.custom.keys()
+        )
+        alignment_keys = set(
+            k for s in sups if s.alignment is not None for k in s.alignment.keys()
+        )
+
+        msup = SupervisionSegment(
+            id=merge(s.id for s in sups),
+            # For MonoCut, the recording_id is always the same.
+            recording_id=sups[0].recording_id,
+            start=mstart,
+            duration=mduration,
+            # For MonoCut, the channel is always the same.
+            channel=sups[0].channel,
+            text=" ".join(s.text for s in sups if s.text) or None,
+            speaker=merge(s.speaker for s in sups if s.speaker) or None,
+            language=merge(s.language for s in sups if s.language) or None,
+            gender=merge(s.gender for s in sups if s.gender) or None,
+            custom={
+                # Merge the string representations of custom fields.
```

Comment:
What kinds of things do the custom fields of Supervision contain?
(I assume the encodings we discussed are in the custom fields of the Cut?)

Reply:
It can be anything really; extra pieces of metadata (e.g. accent, text without normalization), numeric values (snr, some scores), array manifests, really anything. I don’t know of a generic way to support all of that here.. Maybe it should be set to `None` instead?

Reply:
If we end up using this with custom fields we may need a way to do custom types of merge, but I'd say for now this is probably OK.  We'd probably notice if it was wrong.

Reply:
I added a parameter `custom_merge_fn` that can be used to customize this merging behavior, and retained the current code as the default implementation.